packages/mobile/src/screens/create-chat-blast-screen/ChatBlastSelectAudienceFields.tsx

```diff
@@ -100,43 +116,117 @@ const TipSupportersMessageField = () => {
 }
 
 const PastPurchasersMessageField = () => {
-  const { data: user } = useGetCurrentUser()
+  const navigation = useNavigation()
   const [{ value: targetAudience }] = useField(TARGET_AUDIENCE_FIELD)
   const isSelected = targetAudience === ChatBlastAudience.CUSTOMERS
+  const [purchasedContentMetadataField] = useField({
+    name: 'purchased_content_metadata',
+    type: 'select'
+  })
+  const { isDisabled, purchasersCount, premiumContentOptions } =
+    usePurchasersAudience({
+      contentId: purchasedContentMetadataField.value?.contentId,
+      contentType: purchasedContentMetadataField.value?.contentType
+    })
+
+  const handlePress = useCallback(() => {
+    navigation.navigate('ChatBlastSelectContent', {
+      valueName: 'purchased_content_metadata',
+      title: messages.purchasers.placeholder,
+      searchLabel: messages.purchasers.search,
+      content: premiumContentOptions
+    })
+  }, [navigation, premiumContentOptions])
 
   return (
     <ExpandableRadio
       value={ChatBlastAudience.CUSTOMERS}
+      disabled={isDisabled}
       label={
         <LabelWithCount
           label={messages.purchasers.label}
-          // TODO: need purchasers count endpoint
-          count={user?.supporter_count}
+          count={purchasersCount}
           isSelected={isSelected}
         />
       }
       description={messages.purchasers.description}
+      checkedContent={
+        <TouchableOpacity onPress={handlePress}>
+          <Flex
+            row
+            borderTop='default'
+            justifyContent='space-between'
+            alignItems='center'
+            pt='xl' // TODO: should be l
+          >
+            <Text variant='body' size='l' strength='strong'>
+              {messages.purchasers.filterBy}
+            </Text>
+            <IconCaretRight
+              width={spacing.l}
+              height={spacing.l}
+              color='subdued'
+            />
+          </Flex>
+        </TouchableOpacity>
+      }
     />
   )
 }
 
 const RemixCreatorsMessageField = () => {
-  const { data: user } = useGetCurrentUser()
+  const navigation = useNavigation()
   const [{ value: targetAudience }] = useField(TARGET_AUDIENCE_FIELD)
   const isSelected = targetAudience === ChatBlastAudience.REMIXERS
+  const [remixedTrackField] = useField({
+    name: 'remixed_track_id',
+    type: 'select'
+  })
+  const { isDisabled, remixersCount, remixedTracksOptions } =
+    useRemixersAudience({
+      remixedTrackId: remixedTrackField.value?.contentId
+    })
```

Comment:
this is nice! should adopt in web as well

Reply:
i did!